### PR TITLE
Fix issue #26176

### PR DESCRIPTION
**Pull request by andyzheng0831**
This PR is for fixing issue #26176. I verified it by running "kubectl get pods --all-namespaces" to check dns pod. I also ran a failed e2e test case mentioned in the issue to verify.

@yujuhong I assign it to you for review. cc/ @wojtek-t FYI

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -676,9 +676,7 @@ function start-kube-addons {
     fi
     remove-salt-config-comments "${controller_yaml}"
     sed -i -e "s@{{ *base_metrics_memory *}}@${base_metrics_memory}@g" "${controller_yaml}"
-    sed -i -e "s@{{ *metrics_memory *}}@${metrics_memory}@g" "${controller_yaml}"
```
Comment (yujuhong) [inline cluster/gce/gci/configure-helper.sh]: Is it ok to remove these lines? I thought they were here to set the memory requirement for the heapster pods.

Comment (yujuhong) [inline cluster/gce/gci/configure-helper.sh]: Discussed offline. These are not used.

**Comment by yujuhong** [comment]
@mikedanese (oncall), this PR needs to be merged manaully to unblock the submit queue.

**Comment by wojtek-t** [comment]
LGTM

**Comment by mikedanese** [comment]
Ok, What build is this blocking? Should I wait for PR e2e?

**Comment by yujuhong** [comment]
kubernetes-e2e-gke, which is failing at HEAD.

**Comment by mikedanese** [comment]
@k8s-bot unit test this please issue: #25645

**Comment by mikedanese** [comment]
@k8s-bot e2e test this please issue: #26187

**Comment by mikedanese** [comment]
@andyzheng0831 please rebase

**Comment by yujuhong** [comment]
A separate fix (#26195) went in already. Closing this PR.

**Comment by zmerlynn** [comment]
Apologies for slamming the other PR in, but it was blocking .. a lot.

**Comment by yujuhong** [comment]
> Apologies for slamming the other PR in, but it was blocking .. a lot.

Not a problem. Just wanted to point out that the oncall should be donig the manual merges if possible :-)

**Pull request closed without merging**